### ISSUE TITLE
limesurvey: file upload fails

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#86087]
**Describe the bug**

An image upload fails with:

>An error occurred uploading your file. This may be caused by incorrect permissions for the application /tmp folder.

**To Reproduce**
1. create container with limesurvey
```
[root@nixos:~]# nixos-container create foo --config '
services.limesurvey.enable = true;
services.limesurvey.virtualHost =
{
  hostName = "localhost";
  adminAddr = "webmaster@example.org";
};
networking.firewall.allowedTCPPorts = [ 80 ];
'
```
2. start container with `nixos-container start foo`
3. get container IP with `nixos-container show-ip foo`
5. open admin interface at `http://<IP>/admin` and login with `admin/password` 
6. create survey
7. edit survey -> settings -> theme options -> Advanced options -> Upload an image

**Expected behavior**
image upload works

**Screenshots**
![Screenshot from 2020-04-27 02-28-59](https://user-images.githubusercontent.com/91113/80324021-e4663c80-882e-11ea-888a-cc39dcb0e8a3.png)

**Notify maintainers**
@aanderse

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.115, NixOS, 19.09.2407.b67bc34d4e3 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(davidak): `""`
 - channels(root): `"nixos-19.09.2407.b67bc34d4e3, nixos-hardware, nixos-unstable-20.09pre221322.8686922e68d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: services.limesurvey
```

**Comment by aanderse** [comment]
@davidak I'm not a maintainer of this nor listed as one. I think you might have the wrong person.

**Comment by davidak** [comment]
@aanderse yes, but you edited the module before, so i hoped that you can help here.

**Comment by davidak** [comment]
Note:

`"php_admin_value[upload_tmp_dir]" = "${stateDir}/tmp";` does not help

where the hell does `/tmp` come from???

**Comment by davidak** [comment]
@offlinehacker can you continue? i don't know how to fix this.

We would like to use limesurvey for NixCon and the next NixOS release.

**Comment by aanderse** [comment]
@davidak is there a real possibility you'll be using `limesurvey` on `httpd` for NixCon, or just an idea to do so? Who is hosting this instance of `httpd`? Does anyone have any training on building surveys with `limesurvey` (or time/motivation to learn it)?

**Comment by davidak** [comment]
@aanderse i'm working on a community survey with the marketing team the last weeks. that's why one of the nixcon organizers asked me if we have an instance ready to host their survey too. i don't have yet and hoped to be able to fix this. but it would still be OK without or just hack in the nixos logo. the alternative is to use paid hosting, but i would find it kind of embarrassing if nixos is not able to host a basic php app

i'm not a fan of httpd (aka apache2) and i guess nginx is now the preferred webserver for nixos modules, but it should be ok for the time the survey runs

**Comment by aanderse** [comment]
The easiest and quickest course of action may be to configure a NixOS server with `nginx`, `php-fpm`, and `mariadb` then imperatively install `limesurvey` as a regular `php` web application - if you're under time constraints.

**Comment by aanderse** [comment]
After for banging my head on this for a few minutes I'm left to believe this **may** be a bug upstream. To validate it would be easiest to imperatively install and run through the source code. Something like [this](https://github.com/LimeSurvey/LimeSurvey/blob/62d9f87309d0b6d7731ac003408d501a6844b8a3/application/controllers/admin/themes.php#L248) may be the culprit...

**Comment by YaroKasear** [comment]
Has anyone figured out a workaround or a fix for this? I can't import a survey.
